lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -25,58 +28,306 @@
 NIMROD_DEFAULT = -32767.0
 
 TIME_UNIT = cf_units.Unit(
-    "hours since 1970-01-01 00:00:00", calendar=cf_units.CALENDAR_GREGORIAN
+    "seconds since 1970-01-01 00:00:00", calendar=cf_units.CALENDAR_GREGORIAN
 )
 
 
+DEFAULT_UNITS = {
+    817: "m s^-1",
+    804: "knots",
+    422: "min^-1",
+    218: "cm",
+    172: "oktas",
+    155: "m",
+    101: "m",
+    63: "mm hr^-1",
+    61: "mm",
+    58: "Celsius",
+    12: "mb",
+    8: "1",
+}
 FIELD_CODES = {73: "orography"}
-VERTICAL_CODES = {0: "height", 1: "altitude", 12: "levels_below_ground"}
+# VERTICAL_CODES contains conversions from the Nimrod Documentation for the
+# header entry 20 for the vertical coordinate type
+VERTICAL_CODES = {
+    0: {
+        "standard_name": "height",
+        "units": "m",
+        "attributes": {"positive": "up"},
+    },
+    1: {
+        "standard_name": "altitude",
+        "units": "m",
+        "attributes": {"positive": "up"},
+    },
+    2: {
+        "standard_name": "air_pressure",
+        "units": "hPa",
+        "attributes": {"positive": "down"},
+    },
+    6: {"standard_name": "air_temperature", "units": "K"},
+    12: {
+        "long_name": "depth_below_ground",
+        "units": "m",
+        "attributes": {"positive": "down"},
+    },
+}
+# Unhandled VERTICAL_CODES values (no use case identified):
+#  3: ['sigma', 'model level'],
+#  4: ['eta', 'model level'],
+#  5: ['radar beam number', 'unknown'],
+#  7: ['potential temperature', 'unknown'],
+#  8: ['equivalent potential temperature', 'unknown'],
+#  9: ['wet bulb potential temperature', 'unknown'],
+#  10: ['potential vorticity', 'unknown'],
+#  11: ['cloud boundary', 'unknown'],
+
+SOIL_TYPE_CODES = {
+    1: "broadleaf_tree",
+    2: "needleleaf_tree",
+    3: "c3_grass",
+    4: "c4_grass",
+    5: "crop",
+    6: "shrub",
+    7: "urban",
+    8: "water",
+    9: "soil",
+    10: "ice",
+    601: "urban_canyon",
+    602: "urban_roof",
+}
+TIME_AVERAGING_CODES = {
+    8192: "maximum in period",
+    4096: "minimum in period",
+    2048: "unknown(2048)",
+    1024: "unknown(1024)",
+    512: "time lagged",
+    256: "extrapolation",
+    128: "accumulation or average",
+    64: "from UM 150m",
+    32: "scaled to UM resolution",
+    16: "averaged over multiple surface types",
+    8: "only observations used",
+    4: "smoothed",
+    2: "cold bias applied",
+    1: "warm bias applied",
+}
 
 
 class TranslationWarning(Warning):
     pass
 
 
-def name(cube, field):
-    """Set the cube's name from the field."""
+def is_missing(field, value):
+    """Returns True if value matches an "is-missing" number."""
+    return any(
+        np.isclose(value, [field.int_mdi, field.float32_mdi, NIMROD_DEFAULT])
+    )
 
-    cube.rename(field.title.strip())
+
+def name(cube, field):
+    """Set the cube's name from the field.
+    Modifies the Nimrod object title based on other meta-data in the
```

Comment:
The secondary use, of setting/adjusting the field.title, is a bit nasty because it means 'name' must now be run before other routines.
Of course, it's all just splitting out different parts of the 'run' operation.  All the same, the other routines called from 'run; do not modify the field in that way.  It might be clearer to rename this 'set_field_title' or somesuch, and then do the cube name setting in 'run' itself (as it is just one common line).

Reply:
I've removed the logic that updates things on `field` and put the relevant meta-data handling into the correct methods.